### DISCUSSION TITLE
Hidden file contents

### DISCUSSION THREAD

**Discussion by Alexejhero** [community/community#22938] [New to GitHub]
<p>Hello everyone. I was wondering if it was possible to make a file or a folder (mostly a folder) visible on GitHub, but have that file’s contents hidden. Something like when submodules are grey-ed out. </p>
<p>Please excuse my bad english and you should now that I’m new to GitHub.</p>
<p>Thanks in advance</p>

**Comment by ghost** [comment]
<p>Hi <a class="mention" href="/alexejheroytb">@alexejheroytb</a>,</p>
<p>Welcome to the community forums. :slight_smile:</p>
<p>When you say hidden, do you mean greyed out in the file viewer on <a href="http://GitHub.com" rel="nofollow noopener">GitHub.com</a>? If so, there isn’t a way to do that right now, but if instead you simply don’t want a specific file or folder contents to be uploaded to GitHub in the first place, you’ll want to add that file or folder to your .gitignore file</p>
<p>Here are instructions on how to do that: <a href="https://help.github.com/articles/ignoring-files/" rel="nofollow noopener">https://help.github.com/articles/ignoring-files/</a></p>
<p>​If there’s anything else I can help you with, let me know; so happy to help further!</p>
<p>Kindest regards,<br>
Matthew</p>

**Comment by Alexejhero** [comment]
<p>I understand. I wanted to have them grey-ed out so that someone who looks in my repo knows that the folder is there, but they can’t see what’s inside</p>

**Comment by ThirdEyeRose** [comment, marked as answer]
<p>Hi <a class="mention" href="/alexejheroytb">@alexejheroytb</a>, </p>
<p>One of the reasons for this is that if a user can see a repository, they can download or clone that repository as well. Even if a file was hidden in the file viewer, it would be accessible in the downloaded file.</p>
<p>Thanks!</p>

**Comment by Alexejhero** [comment]
<p>I understand. Thank you!</p>
